### ISSUE TITLE
GPUdb.get_records_by_column() method gives error as:  KeyError: 'type_schema'

### ISSUE THREAD

**Issue by SadiCetinkaya** [kineticadb/kinetica-api-python#3]
Hi All,

I try to retrieve some specific columns from a Kineticadb table by the GPUdb.get_records_by_column() method in python. It gives an error as KeyError: 'type_schema'. 
As far as i understand, it's not a directly bug. I only wonder how i can define a schema input for GPUdbRecord.decode_binary_data() function.

`from __future__ import print_function

import collections
import json
import numpy as np
import gpudb
import time

def retrieve():
    KINETICA_HOST = '10.20.10.228'
    KINETICA_PORT = '9191'

    h_db = gpudb.GPUdb(encoding = 'BINARY', host = KINETICA_HOST, port = KINETICA_PORT)

    response = h_db.get_records_by_column(table_name = "udf_json_example",column_names = "domain", offset = 0,  limit = 2,encoding = "binary")
    tablo = gpudb.GPUdbRecord.decode_binary_data(response['type_schema'], response['records_binary'])
    print(tablo)

if __name__ == '__main__':
   a = time.time()
   retrieve()
   b = time.time()
   print(b-a)
`

Here is the error:
`Traceback (most recent call last):
  File "jsonpar.py", line 21, in <module>
    retrieve()
  File "jsonpar.py", line 16, in retrieve
    tablo = gpudb.GPUdbRecord.decode_binary_data(response['type_schema'], response['records_binary'])
KeyError: 'type_schema'
`

**Comment by mmahmud** [comment]
Instead of 'type_schema', try 'response_schema_str'.  'type_schema' is for
GPUdb.get_records() response, whereas GPUdb.get_records_by_column() uses
'response_schema_str'.  Also, to access the records, please use
'binary_encoded_response' ('records_binary' is for get_records() ).  Please
consult
https://www.kinetica.com/docs/api/python/source/gpudb.html#gpudb.GPUdb.get_records_by_column
for further documentation (and
https://www.kinetica.com/docs/api/python/source/gpudb.html#gpudb.GPUdb.get_records
for documentation on get_records()).

Additionally, you may find the GPUdbTable class to be more user friendly.
Please refer to
https://github.com/kineticadb/kinetica-api-python/blob/master/example/example.py
for an example.  When using GPUdbTable (let's say the GPUdbTable object is
named table1), you can retrieve certain column values and have them decoded
in one command:

    records = table1.get_records_by_column( column_names )

where column_names is a list of column names (strings).  You can then
access the column names and values by using items():

    for col, vals in records.items():
        print ("column %s values:" % col, vals)


Please let me know if the above resolves your issues.  Thanks!

**Comment by SadiCetinkaya** [comment]
@mmahmud Also I found json parsing method to retrieve data as in code below:

`response = h_db.get_records_by_column(table_name = "udf_json_example",
column_names = ["domain"], 
offset = 0,  
limit = 8,
encoding = "json")
    d = h_db.parse_dynamic_response(response)['response']['domain']
`
Now I've tried to retrieve 10.000 rows which size is 1,22 MB by both json parsing and get_records_by_column() method. Json seems faster. In anycase, both results are too slow than i expected. Is there any alternative method to read data faster? 

GPUdbTable data read speed: 0.193892002106
json data read speed: 0.100499868393
